src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderPreferencesController.php

```diff
@@ -0,0 +1,111 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Configure\ShopParameters;
+
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Voter\PageVoter;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Controller responsible of "Configure > Shop Parameters > Order Settings" page
+ */
+class OrderPreferencesController extends FrameworkBundleAdminController
+{
+    /**
+     * Show order preferences page
+     *
+     * @param Request $request
+     *
+     * @Template("@PrestaShop/Admin/Configure/ShopParameters/OrderPreferences/order_preferences.html.twig")
+     *
+     * @return array Template parameters
+     */
+    public function indexAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        $form = $this->get('prestashop.admin.order_preferences.form_handler')->getForm();
+
+        return [
+            'layoutTitle' => $this->trans('Order settings', 'Admin.Navigation.Menu'),
+            'requireAddonsSearch' => true,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($legacyController),
+            'form' => $form->createView(),
+            'isMultishippingEnabled' => $this->configuration->getBoolean('PS_ALLOW_MULTISHIPPING'),
+        ];
+    }
+
+    /**
+     * Handle order settings form submit
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        if (!in_array(
+            $this->authorizationLevel($legacyController),
+            [
+                PageVoter::LEVEL_UPDATE,
+                PageVoter::LEVEL_CREATE,
+                PageVoter::LEVEL_DELETE,
+            ]
+        )) {
+            $this->addFlash(
+                'error',
+                $this->trans('You do not have permission to edit this', 'Admin.Notifications.Error')
+            );
+
+            return $this->redirectToRoute('admin_order_preferences');
+        }
+
+        $formHandler = $this->get('prestashop.admin.order_preferences.form_handler');
+
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if ($errors = $formHandler->save($data)) {
```

Comment:
The redirect is the same you can write this part like this to remove duplicate redirect and remove useless variable:
```php
        if ($form->isSubmitted()) {
            if ($errors = $formHandler->save($form->getData())) {
                $this->flashErrors($errors);
            } else {
                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
            }
        }

        return $this->redirectToRoute('admin_order_preferences');
```